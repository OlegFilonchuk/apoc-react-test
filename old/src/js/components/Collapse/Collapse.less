@animationDuration: 0.3s;

.collapse {
  .panel__wrapper {
    overflow: hidden;
    transition: max-height @animationDuration ease-out;

    &--expanded {
      overflow: initial;
      animation-name: expanding;
      animation-duration: @animationDuration;
      animation-delay: 0s;
    }
  }

  .sc-panel-heading {
    position: relative;
  }

  &__custom-content {
    display: flex;
    position: absolute;
    right: 5%;
  }

  &__counter {
    margin-left: 5px;
  }
}

@keyframes expanding {
  from {
    overflow: hidden;
  }
  to {
    overflow: hidden;
  }
}
