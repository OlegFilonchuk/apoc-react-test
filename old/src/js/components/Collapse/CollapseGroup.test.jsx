import React from 'react';
import { mount } from 'enzyme';

import CollapseGroup from './CollapseGroup';
import Collapse from './Collapse';

describe('<CollapseGroup />', () => {
  const collapseIconClass = '.sc-collapse-icon';

  const dummyContent = 'Lorem Ipsum is simply dummy text of the printing and typesetting industry.';

  it('should render CollapseGroup component', () => {
    // arrange
    const firstCollapse = <Collapse title="Title1">{dummyContent}</Collapse>;
    const secondCollapse = <Collapse title="Title2">{dummyContent}</Collapse>;

    // act
    const component = mount(
      <CollapseGroup>
        {firstCollapse}
        {secondCollapse}
      </CollapseGroup>
    );

    // assert
    expect(component).toMatchSnapshot();
  });

  it('should call onToggled when toggling panel', () => {
    // given
    const onToggled = jest.fn();

    const component = mount(
      <CollapseGroup onToggled={onToggled}>
        <Collapse title="Title1">{dummyContent}</Collapse>
        <Collapse title="Title2">{dummyContent}</Collapse>
      </CollapseGroup>
    );

    // when
    component
      .find(Collapse)
      .at(1)
      .find(collapseIconClass)
      .simulate('click');

    // assert
    expect(onToggled).toHaveBeenCalled();
  });

  it('should expand only one Collapse in CollapseGroup component', () => {
    // arrange, act
    const component = mount(
      <CollapseGroup>
        <Collapse title="Title1">{dummyContent}</Collapse>
        <Collapse isCollapsed={false} title="Title2">
          {dummyContent}
        </Collapse>
      </CollapseGroup>
    );

    // assert
    expect(component).toMatchSnapshot();

    const firstCollapse = component.find(Collapse).at(0);
    const secondCollapse = component.find(Collapse).at(1);

    firstCollapse.find(collapseIconClass).simulate('click');
    expect(component).toMatchSnapshot();

    // click on second collapse
    secondCollapse.find(collapseIconClass).simulate('click');
    expect(component).toMatchSnapshot();
  });

  it('should allow to open multiple Collapse components', () => {
    // arrange
    const component = mount(
      <CollapseGroup allowMultipleOpened>
        <Collapse isCollapsed={false} title="Title1">
          {dummyContent}
        </Collapse>
        <Collapse title="Title2">{dummyContent}</Collapse>
        <Collapse isCollapsed={false} title="Title3">
          {dummyContent}
        </Collapse>
      </CollapseGroup>
    );

    // assert
    expect(component).toMatchSnapshot();

    const secondCollapse = component.find(Collapse).at(1);

    secondCollapse.find(collapseIconClass).simulate('click');
    expect(component).toMatchSnapshot();
  });
});
