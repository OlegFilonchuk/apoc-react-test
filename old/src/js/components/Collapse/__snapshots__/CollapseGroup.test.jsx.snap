// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CollapseGroup /> should allow to open multiple Collapse components 1`] = `
<CollapseGroup
  allowMultipleOpened={true}
  className=""
  onToggled={[Function]}
>
  <div
    className="sc-panel-group"
  >
    <Collapse
      className=""
      customContent={null}
      disableShouldComponentUpdate={false}
      displayCounter={false}
      iconPosition="left"
      iconStyles={
        Object {
          "collapsed": "sc-icon-caret-down",
          "expanded": "sc-icon-caret-up",
        }
      }
      id=""
      isClickDisabled={false}
      isCollapsed={false}
      isControlled={false}
      isPanelButton={false}
      key=".0"
      onToggled={[Function]}
      title="Title1"
    >
      <div
        className="sc-panel collapse"
      >
        <CollapseHeader
          className=""
          countSelectedItems={null}
          customContent={null}
          data-test-element="CollapseHeader"
          displayCounter={false}
          iconPosition="left"
          iconStyles={
            Object {
              "collapsed": "sc-icon-caret-down",
              "expanded": "sc-icon-caret-up",
            }
          }
          isClickDisabled={false}
          isCollapsed={false}
          isPanelButton={false}
          onClick={[Function]}
          title="Title1"
        >
          <div
            className="sc-panel-heading"
            data-test-element="CollapseHeader"
            onClick={null}
            role={null}
          >
            <div
              className="sc-panel-title"
            >
              <span
                className="sc-collapse-icon sc-fonticon sc-icon-caret-up"
                onClick={[Function]}
              />
              <h3>
                Title1
              </h3>
            </div>
          </div>
        </CollapseHeader>
        <div
          className="panel__wrapper panel__wrapper--expanded"
          style={
            Object {
              "maxHeight": -1,
            }
          }
        >
          <_class
            offset={true}
            onResize={[Function]}
          >
            <Measure
              contentRect={
                Object {
                  "bounds": Object {},
                  "client": Object {},
                  "entry": Object {},
                  "margin": Object {},
                  "offset": Object {},
                  "scroll": Object {},
                }
              }
              measure={[Function]}
              measureRef={[Function]}
              offset={true}
            >
              <div
                className="sc-panel-body"
              >
                Lorem Ipsum is simply dummy text of the printing and typesetting industry.
              </div>
            </Measure>
          </_class>
        </div>
      </div>
    </Collapse>
    <Collapse
      className=""
      customContent={null}
      disableShouldComponentUpdate={false}
      displayCounter={false}
      iconPosition="left"
      iconStyles={
        Object {
          "collapsed": "sc-icon-caret-down",
          "expanded": "sc-icon-caret-up",
        }
      }
      id=""
      isClickDisabled={false}
      isCollapsed={true}
      isControlled={false}
      isPanelButton={false}
      key=".1"
      onToggled={[Function]}
      title="Title2"
    >
      <div
        className="sc-panel collapse"
      >
        <CollapseHeader
          className=""
          countSelectedItems={null}
          customContent={null}
          data-test-element="CollapseHeader"
          displayCounter={false}
          iconPosition="left"
          iconStyles={
            Object {
              "collapsed": "sc-icon-caret-down",
              "expanded": "sc-icon-caret-up",
            }
          }
          isClickDisabled={false}
          isCollapsed={true}
          isPanelButton={false}
          onClick={[Function]}
          title="Title2"
        >
          <div
            className="sc-panel-heading"
            data-test-element="CollapseHeader"
            onClick={null}
            role={null}
          >
            <div
              className="sc-panel-title"
            >
              <span
                className="sc-collapse-icon sc-fonticon sc-icon-caret-down"
                onClick={[Function]}
              />
              <h3>
                Title2
              </h3>
            </div>
          </div>
        </CollapseHeader>
        <div
          className="panel__wrapper panel__wrapper--collapsed"
          style={
            Object {
              "maxHeight": 0,
            }
          }
        >
          <_class
            offset={true}
            onResize={[Function]}
          >
            <Measure
              contentRect={
                Object {
                  "bounds": Object {},
                  "client": Object {},
                  "entry": Object {},
                  "margin": Object {},
                  "offset": Object {},
                  "scroll": Object {},
                }
              }
              measure={[Function]}
              measureRef={[Function]}
              offset={true}
            >
              <div
                className="sc-panel-body"
              >
                Lorem Ipsum is simply dummy text of the printing and typesetting industry.
              </div>
            </Measure>
          </_class>
        </div>
      </div>
    </Collapse>
    <Collapse
      className=""
      customContent={null}
      disableShouldComponentUpdate={false}
      displayCounter={false}
      iconPosition="left"
      iconStyles={
        Object {
          "collapsed": "sc-icon-caret-down",
          "expanded": "sc-icon-caret-up",
        }
      }
      id=""
      isClickDisabled={false}
      isCollapsed={false}
      isControlled={false}
      isPanelButton={false}
      key=".2"
      onToggled={[Function]}
      title="Title3"
    >
      <div
        className="sc-panel collapse"
      >
        <CollapseHeader
          className=""
          countSelectedItems={null}
          customContent={null}
          data-test-element="CollapseHeader"
          displayCounter={false}
          iconPosition="left"
          iconStyles={
            Object {
              "collapsed": "sc-icon-caret-down",
              "expanded": "sc-icon-caret-up",
            }
          }
          isClickDisabled={false}
          isCollapsed={false}
          isPanelButton={false}
          onClick={[Function]}
          title="Title3"
        >
          <div
            className="sc-panel-heading"
            data-test-element="CollapseHeader"
            onClick={null}
            role={null}
          >
            <div
              className="sc-panel-title"
            >
              <span
                className="sc-collapse-icon sc-fonticon sc-icon-caret-up"
                onClick={[Function]}
              />
              <h3>
                Title3
              </h3>
            </div>
          </div>
        </CollapseHeader>
        <div
          className="panel__wrapper panel__wrapper--expanded"
          style={
            Object {
              "maxHeight": -1,
            }
          }
        >
          <_class
            offset={true}
            onResize={[Function]}
          >
            <Measure
              contentRect={
                Object {
                  "bounds": Object {},
                  "client": Object {},
                  "entry": Object {},
                  "margin": Object {},
                  "offset": Object {},
                  "scroll": Object {},
                }
              }
              measure={[Function]}
              measureRef={[Function]}
              offset={true}
            >
              <div
                className="sc-panel-body"
              >
                Lorem Ipsum is simply dummy text of the printing and typesetting industry.
              </div>
            </Measure>
          </_class>
        </div>
      </div>
    </Collapse>
  </div>
</CollapseGroup>
`;

exports[`<CollapseGroup /> should allow to open multiple Collapse components 2`] = `
<CollapseGroup
  allowMultipleOpened={true}
  className=""
  onToggled={[Function]}
>
  <div
    className="sc-panel-group"
  >
    <Collapse
      className=""
      customContent={null}
      disableShouldComponentUpdate={false}
      displayCounter={false}
      iconPosition="left"
      iconStyles={
        Object {
          "collapsed": "sc-icon-caret-down",
          "expanded": "sc-icon-caret-up",
        }
      }
      id=""
      isClickDisabled={false}
      isCollapsed={false}
      isControlled={false}
      isPanelButton={false}
      key=".0"
      onToggled={[Function]}
      title="Title1"
    >
      <div
        className="sc-panel collapse"
      >
        <CollapseHeader
          className=""
          countSelectedItems={null}
          customContent={null}
          data-test-element="CollapseHeader"
          displayCounter={false}
          iconPosition="left"
          iconStyles={
            Object {
              "collapsed": "sc-icon-caret-down",
              "expanded": "sc-icon-caret-up",
            }
          }
          isClickDisabled={false}
          isCollapsed={false}
          isPanelButton={false}
          onClick={[Function]}
          title="Title1"
        >
          <div
            className="sc-panel-heading"
            data-test-element="CollapseHeader"
            onClick={null}
            role={null}
          >
            <div
              className="sc-panel-title"
            >
              <span
                className="sc-collapse-icon sc-fonticon sc-icon-caret-up"
                onClick={[Function]}
              />
              <h3>
                Title1
              </h3>
            </div>
          </div>
        </CollapseHeader>
        <div
          className="panel__wrapper panel__wrapper--expanded"
          style={
            Object {
              "maxHeight": -1,
            }
          }
        >
          <_class
            offset={true}
            onResize={[Function]}
          >
            <Measure
              contentRect={
                Object {
                  "bounds": Object {},
                  "client": Object {},
                  "entry": Object {},
                  "margin": Object {},
                  "offset": Object {},
                  "scroll": Object {},
                }
              }
              measure={[Function]}
              measureRef={[Function]}
              offset={true}
            >
              <div
                className="sc-panel-body"
              >
                Lorem Ipsum is simply dummy text of the printing and typesetting industry.
              </div>
            </Measure>
          </_class>
        </div>
      </div>
    </Collapse>
    <Collapse
      className=""
      customContent={null}
      disableShouldComponentUpdate={false}
      displayCounter={false}
      iconPosition="left"
      iconStyles={
        Object {
          "collapsed": "sc-icon-caret-down",
          "expanded": "sc-icon-caret-up",
        }
      }
      id=""
      isClickDisabled={false}
      isCollapsed={false}
      isControlled={false}
      isPanelButton={false}
      key=".1"
      onToggled={[Function]}
      title="Title2"
    >
      <div
        className="sc-panel collapse"
      >
        <CollapseHeader
          className=""
          countSelectedItems={null}
          customContent={null}
          data-test-element="CollapseHeader"
          displayCounter={false}
          iconPosition="left"
          iconStyles={
            Object {
              "collapsed": "sc-icon-caret-down",
              "expanded": "sc-icon-caret-up",
            }
          }
          isClickDisabled={false}
          isCollapsed={false}
          isPanelButton={false}
          onClick={[Function]}
          title="Title2"
        >
          <div
            className="sc-panel-heading"
            data-test-element="CollapseHeader"
            onClick={null}
            role={null}
          >
            <div
              className="sc-panel-title"
            >
              <span
                className="sc-collapse-icon sc-fonticon sc-icon-caret-up"
                onClick={[Function]}
              />
              <h3>
                Title2
              </h3>
            </div>
          </div>
        </CollapseHeader>
        <div
          className="panel__wrapper panel__wrapper--expanded"
          style={
            Object {
              "maxHeight": -1,
            }
          }
        >
          <_class
            offset={true}
            onResize={[Function]}
          >
            <Measure
              contentRect={
                Object {
                  "bounds": Object {},
                  "client": Object {},
                  "entry": Object {},
                  "margin": Object {},
                  "offset": Object {},
                  "scroll": Object {},
                }
              }
              measure={[Function]}
              measureRef={[Function]}
              offset={true}
            >
              <div
                className="sc-panel-body"
              >
                Lorem Ipsum is simply dummy text of the printing and typesetting industry.
              </div>
            </Measure>
          </_class>
        </div>
      </div>
    </Collapse>
    <Collapse
      className=""
      customContent={null}
      disableShouldComponentUpdate={false}
      displayCounter={false}
      iconPosition="left"
      iconStyles={
        Object {
          "collapsed": "sc-icon-caret-down",
          "expanded": "sc-icon-caret-up",
        }
      }
      id=""
      isClickDisabled={false}
      isCollapsed={false}
      isControlled={false}
      isPanelButton={false}
      key=".2"
      onToggled={[Function]}
      title="Title3"
    >
      <div
        className="sc-panel collapse"
      >
        <CollapseHeader
          className=""
          countSelectedItems={null}
          customContent={null}
          data-test-element="CollapseHeader"
          displayCounter={false}
          iconPosition="left"
          iconStyles={
            Object {
              "collapsed": "sc-icon-caret-down",
              "expanded": "sc-icon-caret-up",
            }
          }
          isClickDisabled={false}
          isCollapsed={false}
          isPanelButton={false}
          onClick={[Function]}
          title="Title3"
        >
          <div
            className="sc-panel-heading"
            data-test-element="CollapseHeader"
            onClick={null}
            role={null}
          >
            <div
              className="sc-panel-title"
            >
              <span
                className="sc-collapse-icon sc-fonticon sc-icon-caret-up"
                onClick={[Function]}
              />
              <h3>
                Title3
              </h3>
            </div>
          </div>
        </CollapseHeader>
        <div
          className="panel__wrapper panel__wrapper--expanded"
          style={
            Object {
              "maxHeight": -1,
            }
          }
        >
          <_class
            offset={true}
            onResize={[Function]}
          >
            <Measure
              contentRect={
                Object {
                  "bounds": Object {},
                  "client": Object {},
                  "entry": Object {},
                  "margin": Object {},
                  "offset": Object {},
                  "scroll": Object {},
                }
              }
              measure={[Function]}
              measureRef={[Function]}
              offset={true}
            >
              <div
                className="sc-panel-body"
              >
                Lorem Ipsum is simply dummy text of the printing and typesetting industry.
              </div>
            </Measure>
          </_class>
        </div>
      </div>
    </Collapse>
  </div>
</CollapseGroup>
`;

exports[`<CollapseGroup /> should expand only one Collapse in CollapseGroup component 1`] = `
<CollapseGroup
  allowMultipleOpened={false}
  className=""
  onToggled={[Function]}
>
  <div
    className="sc-panel-group"
  >
    <Collapse
      className=""
      customContent={null}
      disableShouldComponentUpdate={false}
      displayCounter={false}
      iconPosition="left"
      iconStyles={
        Object {
          "collapsed": "sc-icon-caret-down",
          "expanded": "sc-icon-caret-up",
        }
      }
      id=""
      isClickDisabled={false}
      isCollapsed={true}
      isControlled={false}
      isPanelButton={false}
      key=".0"
      onToggled={[Function]}
      title="Title1"
    >
      <div
        className="sc-panel collapse"
      >
        <CollapseHeader
          className=""
          countSelectedItems={null}
          customContent={null}
          data-test-element="CollapseHeader"
          displayCounter={false}
          iconPosition="left"
          iconStyles={
            Object {
              "collapsed": "sc-icon-caret-down",
              "expanded": "sc-icon-caret-up",
            }
          }
          isClickDisabled={false}
          isCollapsed={true}
          isPanelButton={false}
          onClick={[Function]}
          title="Title1"
        >
          <div
            className="sc-panel-heading"
            data-test-element="CollapseHeader"
            onClick={null}
            role={null}
          >
            <div
              className="sc-panel-title"
            >
              <span
                className="sc-collapse-icon sc-fonticon sc-icon-caret-down"
                onClick={[Function]}
              />
              <h3>
                Title1
              </h3>
            </div>
          </div>
        </CollapseHeader>
        <div
          className="panel__wrapper panel__wrapper--collapsed"
          style={
            Object {
              "maxHeight": 0,
            }
          }
        >
          <_class
            offset={true}
            onResize={[Function]}
          >
            <Measure
              contentRect={
                Object {
                  "bounds": Object {},
                  "client": Object {},
                  "entry": Object {},
                  "margin": Object {},
                  "offset": Object {},
                  "scroll": Object {},
                }
              }
              measure={[Function]}
              measureRef={[Function]}
              offset={true}
            >
              <div
                className="sc-panel-body"
              >
                Lorem Ipsum is simply dummy text of the printing and typesetting industry.
              </div>
            </Measure>
          </_class>
        </div>
      </div>
    </Collapse>
    <Collapse
      className=""
      customContent={null}
      disableShouldComponentUpdate={false}
      displayCounter={false}
      iconPosition="left"
      iconStyles={
        Object {
          "collapsed": "sc-icon-caret-down",
          "expanded": "sc-icon-caret-up",
        }
      }
      id=""
      isClickDisabled={false}
      isCollapsed={false}
      isControlled={false}
      isPanelButton={false}
      key=".1"
      onToggled={[Function]}
      title="Title2"
    >
      <div
        className="sc-panel collapse"
      >
        <CollapseHeader
          className=""
          countSelectedItems={null}
          customContent={null}
          data-test-element="CollapseHeader"
          displayCounter={false}
          iconPosition="left"
          iconStyles={
            Object {
              "collapsed": "sc-icon-caret-down",
              "expanded": "sc-icon-caret-up",
            }
          }
          isClickDisabled={false}
          isCollapsed={false}
          isPanelButton={false}
          onClick={[Function]}
          title="Title2"
        >
          <div
            className="sc-panel-heading"
            data-test-element="CollapseHeader"
            onClick={null}
            role={null}
          >
            <div
              className="sc-panel-title"
            >
              <span
                className="sc-collapse-icon sc-fonticon sc-icon-caret-up"
                onClick={[Function]}
              />
              <h3>
                Title2
              </h3>
            </div>
          </div>
        </CollapseHeader>
        <div
          className="panel__wrapper panel__wrapper--expanded"
          style={
            Object {
              "maxHeight": -1,
            }
          }
        >
          <_class
            offset={true}
            onResize={[Function]}
          >
            <Measure
              contentRect={
                Object {
                  "bounds": Object {},
                  "client": Object {},
                  "entry": Object {},
                  "margin": Object {},
                  "offset": Object {},
                  "scroll": Object {},
                }
              }
              measure={[Function]}
              measureRef={[Function]}
              offset={true}
            >
              <div
                className="sc-panel-body"
              >
                Lorem Ipsum is simply dummy text of the printing and typesetting industry.
              </div>
            </Measure>
          </_class>
        </div>
      </div>
    </Collapse>
  </div>
</CollapseGroup>
`;

exports[`<CollapseGroup /> should expand only one Collapse in CollapseGroup component 2`] = `
<CollapseGroup
  allowMultipleOpened={false}
  className=""
  onToggled={[Function]}
>
  <div
    className="sc-panel-group"
  >
    <Collapse
      className=""
      customContent={null}
      disableShouldComponentUpdate={false}
      displayCounter={false}
      iconPosition="left"
      iconStyles={
        Object {
          "collapsed": "sc-icon-caret-down",
          "expanded": "sc-icon-caret-up",
        }
      }
      id=""
      isClickDisabled={false}
      isCollapsed={false}
      isControlled={false}
      isPanelButton={false}
      key=".0"
      onToggled={[Function]}
      title="Title1"
    >
      <div
        className="sc-panel collapse"
      >
        <CollapseHeader
          className=""
          countSelectedItems={null}
          customContent={null}
          data-test-element="CollapseHeader"
          displayCounter={false}
          iconPosition="left"
          iconStyles={
            Object {
              "collapsed": "sc-icon-caret-down",
              "expanded": "sc-icon-caret-up",
            }
          }
          isClickDisabled={false}
          isCollapsed={false}
          isPanelButton={false}
          onClick={[Function]}
          title="Title1"
        >
          <div
            className="sc-panel-heading"
            data-test-element="CollapseHeader"
            onClick={null}
            role={null}
          >
            <div
              className="sc-panel-title"
            >
              <span
                className="sc-collapse-icon sc-fonticon sc-icon-caret-up"
                onClick={[Function]}
              />
              <h3>
                Title1
              </h3>
            </div>
          </div>
        </CollapseHeader>
        <div
          className="panel__wrapper panel__wrapper--expanded"
          style={
            Object {
              "maxHeight": -1,
            }
          }
        >
          <_class
            offset={true}
            onResize={[Function]}
          >
            <Measure
              contentRect={
                Object {
                  "bounds": Object {},
                  "client": Object {},
                  "entry": Object {},
                  "margin": Object {},
                  "offset": Object {},
                  "scroll": Object {},
                }
              }
              measure={[Function]}
              measureRef={[Function]}
              offset={true}
            >
              <div
                className="sc-panel-body"
              >
                Lorem Ipsum is simply dummy text of the printing and typesetting industry.
              </div>
            </Measure>
          </_class>
        </div>
      </div>
    </Collapse>
    <Collapse
      className=""
      customContent={null}
      disableShouldComponentUpdate={false}
      displayCounter={false}
      iconPosition="left"
      iconStyles={
        Object {
          "collapsed": "sc-icon-caret-down",
          "expanded": "sc-icon-caret-up",
        }
      }
      id=""
      isClickDisabled={false}
      isCollapsed={true}
      isControlled={false}
      isPanelButton={false}
      key=".1"
      onToggled={[Function]}
      title="Title2"
    >
      <div
        className="sc-panel collapse"
      >
        <CollapseHeader
          className=""
          countSelectedItems={null}
          customContent={null}
          data-test-element="CollapseHeader"
          displayCounter={false}
          iconPosition="left"
          iconStyles={
            Object {
              "collapsed": "sc-icon-caret-down",
              "expanded": "sc-icon-caret-up",
            }
          }
          isClickDisabled={false}
          isCollapsed={true}
          isPanelButton={false}
          onClick={[Function]}
          title="Title2"
        >
          <div
            className="sc-panel-heading"
            data-test-element="CollapseHeader"
            onClick={null}
            role={null}
          >
            <div
              className="sc-panel-title"
            >
              <span
                className="sc-collapse-icon sc-fonticon sc-icon-caret-down"
                onClick={[Function]}
              />
              <h3>
                Title2
              </h3>
            </div>
          </div>
        </CollapseHeader>
        <div
          className="panel__wrapper panel__wrapper--collapsed"
          style={
            Object {
              "maxHeight": 0,
            }
          }
        >
          <_class
            offset={true}
            onResize={[Function]}
          >
            <Measure
              contentRect={
                Object {
                  "bounds": Object {},
                  "client": Object {},
                  "entry": Object {},
                  "margin": Object {},
                  "offset": Object {},
                  "scroll": Object {},
                }
              }
              measure={[Function]}
              measureRef={[Function]}
              offset={true}
            >
              <div
                className="sc-panel-body"
              >
                Lorem Ipsum is simply dummy text of the printing and typesetting industry.
              </div>
            </Measure>
          </_class>
        </div>
      </div>
    </Collapse>
  </div>
</CollapseGroup>
`;

exports[`<CollapseGroup /> should expand only one Collapse in CollapseGroup component 3`] = `
<CollapseGroup
  allowMultipleOpened={false}
  className=""
  onToggled={[Function]}
>
  <div
    className="sc-panel-group"
  >
    <Collapse
      className=""
      customContent={null}
      disableShouldComponentUpdate={false}
      displayCounter={false}
      iconPosition="left"
      iconStyles={
        Object {
          "collapsed": "sc-icon-caret-down",
          "expanded": "sc-icon-caret-up",
        }
      }
      id=""
      isClickDisabled={false}
      isCollapsed={true}
      isControlled={false}
      isPanelButton={false}
      key=".0"
      onToggled={[Function]}
      title="Title1"
    >
      <div
        className="sc-panel collapse"
      >
        <CollapseHeader
          className=""
          countSelectedItems={null}
          customContent={null}
          data-test-element="CollapseHeader"
          displayCounter={false}
          iconPosition="left"
          iconStyles={
            Object {
              "collapsed": "sc-icon-caret-down",
              "expanded": "sc-icon-caret-up",
            }
          }
          isClickDisabled={false}
          isCollapsed={true}
          isPanelButton={false}
          onClick={[Function]}
          title="Title1"
        >
          <div
            className="sc-panel-heading"
            data-test-element="CollapseHeader"
            onClick={null}
            role={null}
          >
            <div
              className="sc-panel-title"
            >
              <span
                className="sc-collapse-icon sc-fonticon sc-icon-caret-down"
                onClick={[Function]}
              />
              <h3>
                Title1
              </h3>
            </div>
          </div>
        </CollapseHeader>
        <div
          className="panel__wrapper panel__wrapper--collapsed"
          style={
            Object {
              "maxHeight": 0,
            }
          }
        >
          <_class
            offset={true}
            onResize={[Function]}
          >
            <Measure
              contentRect={
                Object {
                  "bounds": Object {},
                  "client": Object {},
                  "entry": Object {},
                  "margin": Object {},
                  "offset": Object {},
                  "scroll": Object {},
                }
              }
              measure={[Function]}
              measureRef={[Function]}
              offset={true}
            >
              <div
                className="sc-panel-body"
              >
                Lorem Ipsum is simply dummy text of the printing and typesetting industry.
              </div>
            </Measure>
          </_class>
        </div>
      </div>
    </Collapse>
    <Collapse
      className=""
      customContent={null}
      disableShouldComponentUpdate={false}
      displayCounter={false}
      iconPosition="left"
      iconStyles={
        Object {
          "collapsed": "sc-icon-caret-down",
          "expanded": "sc-icon-caret-up",
        }
      }
      id=""
      isClickDisabled={false}
      isCollapsed={false}
      isControlled={false}
      isPanelButton={false}
      key=".1"
      onToggled={[Function]}
      title="Title2"
    >
      <div
        className="sc-panel collapse"
      >
        <CollapseHeader
          className=""
          countSelectedItems={null}
          customContent={null}
          data-test-element="CollapseHeader"
          displayCounter={false}
          iconPosition="left"
          iconStyles={
            Object {
              "collapsed": "sc-icon-caret-down",
              "expanded": "sc-icon-caret-up",
            }
          }
          isClickDisabled={false}
          isCollapsed={false}
          isPanelButton={false}
          onClick={[Function]}
          title="Title2"
        >
          <div
            className="sc-panel-heading"
            data-test-element="CollapseHeader"
            onClick={null}
            role={null}
          >
            <div
              className="sc-panel-title"
            >
              <span
                className="sc-collapse-icon sc-fonticon sc-icon-caret-up"
                onClick={[Function]}
              />
              <h3>
                Title2
              </h3>
            </div>
          </div>
        </CollapseHeader>
        <div
          className="panel__wrapper panel__wrapper--expanded"
          style={
            Object {
              "maxHeight": -1,
            }
          }
        >
          <_class
            offset={true}
            onResize={[Function]}
          >
            <Measure
              contentRect={
                Object {
                  "bounds": Object {},
                  "client": Object {},
                  "entry": Object {},
                  "margin": Object {},
                  "offset": Object {},
                  "scroll": Object {},
                }
              }
              measure={[Function]}
              measureRef={[Function]}
              offset={true}
            >
              <div
                className="sc-panel-body"
              >
                Lorem Ipsum is simply dummy text of the printing and typesetting industry.
              </div>
            </Measure>
          </_class>
        </div>
      </div>
    </Collapse>
  </div>
</CollapseGroup>
`;

exports[`<CollapseGroup /> should render CollapseGroup component 1`] = `
<CollapseGroup
  allowMultipleOpened={false}
  className=""
  onToggled={[Function]}
>
  <div
    className="sc-panel-group"
  >
    <Collapse
      className=""
      customContent={null}
      disableShouldComponentUpdate={false}
      displayCounter={false}
      iconPosition="left"
      iconStyles={
        Object {
          "collapsed": "sc-icon-caret-down",
          "expanded": "sc-icon-caret-up",
        }
      }
      id=""
      isClickDisabled={false}
      isCollapsed={true}
      isControlled={false}
      isPanelButton={false}
      key=".0"
      onToggled={[Function]}
      title="Title1"
    >
      <div
        className="sc-panel collapse"
      >
        <CollapseHeader
          className=""
          countSelectedItems={null}
          customContent={null}
          data-test-element="CollapseHeader"
          displayCounter={false}
          iconPosition="left"
          iconStyles={
            Object {
              "collapsed": "sc-icon-caret-down",
              "expanded": "sc-icon-caret-up",
            }
          }
          isClickDisabled={false}
          isCollapsed={true}
          isPanelButton={false}
          onClick={[Function]}
          title="Title1"
        >
          <div
            className="sc-panel-heading"
            data-test-element="CollapseHeader"
            onClick={null}
            role={null}
          >
            <div
              className="sc-panel-title"
            >
              <span
                className="sc-collapse-icon sc-fonticon sc-icon-caret-down"
                onClick={[Function]}
              />
              <h3>
                Title1
              </h3>
            </div>
          </div>
        </CollapseHeader>
        <div
          className="panel__wrapper panel__wrapper--collapsed"
          style={
            Object {
              "maxHeight": 0,
            }
          }
        >
          <_class
            offset={true}
            onResize={[Function]}
          >
            <Measure
              contentRect={
                Object {
                  "bounds": Object {},
                  "client": Object {},
                  "entry": Object {},
                  "margin": Object {},
                  "offset": Object {},
                  "scroll": Object {},
                }
              }
              measure={[Function]}
              measureRef={[Function]}
              offset={true}
            >
              <div
                className="sc-panel-body"
              >
                Lorem Ipsum is simply dummy text of the printing and typesetting industry.
              </div>
            </Measure>
          </_class>
        </div>
      </div>
    </Collapse>
    <Collapse
      className=""
      customContent={null}
      disableShouldComponentUpdate={false}
      displayCounter={false}
      iconPosition="left"
      iconStyles={
        Object {
          "collapsed": "sc-icon-caret-down",
          "expanded": "sc-icon-caret-up",
        }
      }
      id=""
      isClickDisabled={false}
      isCollapsed={true}
      isControlled={false}
      isPanelButton={false}
      key=".1"
      onToggled={[Function]}
      title="Title2"
    >
      <div
        className="sc-panel collapse"
      >
        <CollapseHeader
          className=""
          countSelectedItems={null}
          customContent={null}
          data-test-element="CollapseHeader"
          displayCounter={false}
          iconPosition="left"
          iconStyles={
            Object {
              "collapsed": "sc-icon-caret-down",
              "expanded": "sc-icon-caret-up",
            }
          }
          isClickDisabled={false}
          isCollapsed={true}
          isPanelButton={false}
          onClick={[Function]}
          title="Title2"
        >
          <div
            className="sc-panel-heading"
            data-test-element="CollapseHeader"
            onClick={null}
            role={null}
          >
            <div
              className="sc-panel-title"
            >
              <span
                className="sc-collapse-icon sc-fonticon sc-icon-caret-down"
                onClick={[Function]}
              />
              <h3>
                Title2
              </h3>
            </div>
          </div>
        </CollapseHeader>
        <div
          className="panel__wrapper panel__wrapper--collapsed"
          style={
            Object {
              "maxHeight": 0,
            }
          }
        >
          <_class
            offset={true}
            onResize={[Function]}
          >
            <Measure
              contentRect={
                Object {
                  "bounds": Object {},
                  "client": Object {},
                  "entry": Object {},
                  "margin": Object {},
                  "offset": Object {},
                  "scroll": Object {},
                }
              }
              measure={[Function]}
              measureRef={[Function]}
              offset={true}
            >
              <div
                className="sc-panel-body"
              >
                Lorem Ipsum is simply dummy text of the printing and typesetting industry.
              </div>
            </Measure>
          </_class>
        </div>
      </div>
    </Collapse>
  </div>
</CollapseGroup>
`;
